src/cpp/session/projects/SessionProjects.cpp

```diff
@@ -178,27 +208,31 @@ Error createProject(const json::JsonRpcRequest& request,
                                       ProjectContext::buildDefaults(),
                                       ProjectContext::defaultConfig());
    }
-
-   // default project
-   else
+   
+   // if we have a custom project template, call that first
+   if (!projectTemplateOptions.is_null())
```

Comment:
for sanity, consider making sure its type is `json::Object` here too (otherwise it looks like `initializeProjectFromTemplate` can throw)